libraries/wallet/wallet.cpp

```diff
@@ -808,10 +808,25 @@ class wallet_api_impl
          //
          // http://en.wikipedia.org/wiki/Most_vexing_parse
          //
-         fc::ofstream outfile{ fc::path( wallet_filename ) };
+         fc::ofstream outfile{ fc::path( wallet_filename + ".tmp" ) };
          outfile.write( data.c_str(), data.length() );
          outfile.flush();
          outfile.close();
+
+         std::string wallet_file_content;
+         fc::read_file_contents(wallet_filename, wallet_file_content);
+
+         if (wallet_file_content == data) {
+            fc::rename( wallet_filename + ".tmp", wallet_filename );
+         } 
+         else 
+         {
+            disable_umask_protection();
```

Comment:
This is not necessary, because it already happens in the outer try/catch.

Reply:
Why it is not necessary ? I've tested & checked it and we won't get into outer try/catch code after throwing exception here that's why I've placed `disable_umask_protection`

Reply:
Using throw without specifying an exception will terminate instead of actually throwing, see below.

Reply:
yes, you're absolutely right ! I've checked it earlier, that's why I told you that `disable_umask_protection` is necessary there before using `throw`, if I will change `throw` to `fc::exception` subtype then I will remove `disable_umask_protection` call. Just couldn't understand why you told that `disable_umask_protection` isn't necessary before using `throw`.